### DISCUSSION TITLE
Integrating three-vrm with AR in React Native - Advice Needed

### DISCUSSION THREAD

**Discussion by tonyabracadabra** [pixiv/three-vrm#1236] [Help]
Hey,

First off, kudos to you guys for the cool open-source project, three-vrm. I've been messing around with it and it's great stuff.

Now, I'm trying to pull three-vrm into a React Native AR app. The goal? Drop an AR model right into a real-world scene. I get that three-vrm was mostly made with web apps in mind, but I'd love to see it rock in a React Native AR app.

To make that happen, I'm eyeing a few libraries and tools, like react-three-fiber, to help with creating 3D scenes and interactions. But there might be better choices out there for AR integration with React Native and three-vrm.

Here's my game plan so far:

Pull in the necessary stuff from three and @pixiv/three-vrm.
Load the AR model with GLTFLoader.
Get the VRMLoaderPlugin to load the AR model into an AR scene.
Display the AR model in the AR scene in the React Native app.
I know this might not be a cakewalk. I might run into performance issues, compatibility hiccups, and other roadblocks. So, I'm here for your advice. Do you guys recommend any specific libraries or tools for making three-vrm play nice with AR in React Native?

Any pointers or tips would be a huge help.

Thanks for your time and for the awesome work you're doing with three-vrm.

Cheers

**Comment by ke456-png** [comment]
Hi @tonyabracadabra

I think it's a great idea to combine react-three-fiber and three-vrm, although I haven't tested it on React Native!
However, I don't think it is easy to work with native AR apps.

Alternatively,  Unity + UniVRM is another way.
